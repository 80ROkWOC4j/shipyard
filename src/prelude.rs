pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
