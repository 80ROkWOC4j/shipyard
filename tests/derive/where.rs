use shipyard::prelude::*;

#[system(Test)]
fn run(_: &usize) where usize: Debug {}

fn main() {}
