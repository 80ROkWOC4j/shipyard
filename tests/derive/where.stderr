error: custom attribute panicked
 --> $DIR/where.rs:3:1
  |
3 | #[system(Test)]
  | ^^^^^^^^^^^^^^^
  |
  = help: message: run should not take generic arguments nor where clause.
